File: Exercises/Exercise1/excercise1.py
# Python 3.3.3 and 2.7.6
# python helloworld_python.py

from threading import Thread

i = 0

def thread_func1():
    global i
    for x in xrange(1,1000000):
        i=i+1

def thread_func2():
    global i
    for x in xrange(1,1000000):
        i=i-1

def main():
    thread1 = Thread(target = thread_func1, args = (),)
    thread2 = Thread(target = thread_func2, args = (),)
    thread1.start()
    thread2.start()

    thread2.join()
    thread1.join()
    print("This is the value of i: {0}".format(i))


main()

File: Exercises/Exercise1/excercise1.c
#include<pthread.h>
#include<stdio.h>

#define MAX 1000000
int i = 0;

void* thread_func1(){
    for(int x=0; x < MAX; x++){
        i++;
    }
    return NULL;
}

void* thread_func2(){
    for(int x=0; x < MAX; x++){
        i--;
    }
    return NULL;
}

int main(){

    pthread_t thread1;
    pthread_t thread2;
    pthread_create(&thread1, NULL, thread_func1, NULL);
    pthread_create(&thread2, NULL, thread_func2, NULL);
    pthread_join(thread1, NULL);
    pthread_join(thread2, NULL);
    printf("Hey this is i: %i",i);

    return 0;
}

